### PR TITLE
Fix timeout in process actions

### PR DESCRIPTION
**Pull request by superstar54**
Issue: My terminal hangs when I run `verdi process play pk` for an unreachable process.

Reason:
When a future is finished, it can be `done` or `excepted`. If the future is excepted, one will wait forever using `future.result()`.
Also, it is good to add timeout in the `exception` and `result`.

### CODE REVIEW

File: aiida/engine/processes/control.py [hunk 1]
```diff
@@ -218,6 +218,7 @@ def _perform_actions(
     _resolve_futures(futures, infinitive, present, wait, timeout)
 
 
+#pylint: disable=too-many-branches
```
Comment (sphuber) [inline aiida/engine/processes/control.py:221]: This is now actually disabling for all of the rest of the file. You should move it next to the function definition, e.g.
```python
def _resolve_futures(  # pylint: disable=too-many-branches
```

**Review by sphuber — CHANGES_REQUESTED** [review]
Could you please add a test for this. You would have to create a process without the task being sent to RabbitMQ. You can see an example of how to do this in `tests/engine/processes/test_control.py` in the `test_revive` test:
```python

    # Temporarily patch the ``RemoteProcessThreadController.continue_process`` method to do nothing and just return a
    # completed future. This ensures that the submission creates a process node in the database but no task is sent to
    # RabbitMQ and so the daemon will not start running it.
    with monkeypatch.context() as context:
        context.setattr(RemoteProcessThreadController, 'continue_process', lambda *args, **kwargs: None)
        node = submit(builder)
```
you can then call `_resolve_futures` (or whatever the parent function is) with a timeout and check that it indeed timesout.

**Comment by sphuber** [comment]
@superstar54 what is the plan for this PR? Do you still want to have a look at adding a test?

**Comment by sphuber** [comment]
pinging @superstar54

**Comment by superstar54** [comment]
@sphuber, I did some tests and found the original code handles the timeout correctly. And I don't remember how to reproduce the error. So I am closing this PR now and will reopen it when I encounter the issue again.

**Pull request closed without merging**